Zend/zend_variables.c

```diff
@@ -51,6 +51,10 @@ static const zend_rc_dtor_func_t zend_rc_dtor_func[] = {
 	[IS_CONSTANT_AST] = (zend_rc_dtor_func_t)zend_ast_ref_destroy
 };
 
+#if ZEND_GCC_VERSION >= 4006
```

Comment:
The checks is still wrong tough as I ve tested your branch on centos 7 (they are users with this this still).

Reply:
At least a configure check in the proper *.m4 file would be better.

Reply:
Adding a configure check will just slow down configure for something that has no runtime effect. I don't think that's worth it.
Before I wrote this `#if`, I consulted https://gcc.gnu.org/wiki/C11Status and it said GCC 4.6 has C11 static assertions (I didn't actually check with 4.6 because I don't have that version anywhere).
If you have other information, please share details. You said you tested with CentOS 7, but you didn't say what the test result was.

Reply:
is it possible for you to try locally with clang if it reaches that code ?

Reply:
It's not because clang pretends to be GCC 4.2. (Tested with clang 14 and 16)
Do you want me to add a test for clang? clang appears to implement static assertions in all versions according to https://clang.llvm.org/c_status.html#c11

Reply:
Sure why not.

Reply:
Added.
(I didn't initially do this because I wanted to keep this as simple as possible; and having this assertion only in the CI would have been enough to detect mistakes prior to merging; unfortunately there's no way to detect the availability of this language feature with `__has_feature` at compile time, or I missed it.)